src/transactions/transactions.controller.ts

```diff
@@ -13,23 +13,154 @@ import { TransactionsService } from './transactions.service';
 import { CreateTransactionDto } from './dto/create-transaction.dto';
 import { UpdateTransactionDto } from './dto/update-transaction.dto';
 import { AuthGuard } from '../auth/auth.guard';
+import { ApiResponse } from '@nestjs/swagger';
 
 @Controller('transactions')
 export class TransactionsController {
   constructor(private readonly transactionsService: TransactionsService) {}
 
   @Post()
   @UseGuards(AuthGuard)
-  create(@Body() createTransactionDto: CreateTransactionDto) {
+  @ApiResponse({
+    status: 201,
+    description: 'Transactions created successfully.',
+    schema: {
+      type: 'object',
+      properties: {
+        status: {
+          type: 'string',
+          example: 'success',
+        },
+        message: {
+          type: 'string',
+          example: 'Transactions created successfully',
+        },
+        data: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              transaction_id: {
+                type: 'string',
+                format: 'uuid',
+                example: '123e4567-e89b-12d3-a456-426614174000',
+              },
+              title: {
+                type: 'string',
+                example: 'pecel lele',
+              },
+              amount: {
+                type: 'string',
+                example: '30000',
+              },
+              type: {
+                type: 'string',
+                example: 'debt',
+              },
+              category: {
+                type: 'string',
+                example: 'food',
+              },
+              currency: {
+                type: 'string',
+                example: 'IDR',
+              },
+              user_id: {
+                type: 'string',
+                format: 'uuid',
+                example: '456e1234-e89b-12d3-a456-426614174000',
+              },
+              createdAt: {
+                type: 'string',
+                format: 'date-time',
+                example: '2024-11-17T12:00:00Z',
+              },
+              updatedAt: {
+                type: 'string',
+                format: 'date-time',
+                example: '2024-11-17T13:00:00Z',
+              },
+            },
+          },
+        },
+      },
+    },
+  })
+  create(@Body() payload: { user_id: string; data: CreateTransactionDto[] }) {
```

Comment:
i think its better to change CreateTransactioDto instead using payload, you can check this reference
1. https://chatgpt.com/share/6741c844-83e8-8006-adf7-0853dd713b59
2. https://www.reddit.com/r/Nestjs_framework/comments/r1p9qc/how_to_expose_an_array_of_objects_using_a_dto/

Reply:
woww nice idea, ok i've changed it, sungkem🙏🏻 senior dev